File: src/api/responses/ErrorResponse.cs
using System;

namespace DVAN.API
{
    /// <summary>
    /// Error response containing the request endpoint and a error message.
    /// </summary>
    class ErrorResponse
    {
        /// <summary>
        /// The request endpoint.
        /// </summary>
        /// <example>api/endpoint/action</example>
        public string request { get; set; }

        /// <summary>
        /// The error message.
        /// </summary>
        /// <example>invalid request</example>
        public string error { get; set; }

        public ErrorResponse(string request, string message)
        {
            this.request = request;
            this.error = message;
        }
    }
}

File: src/main/java/org/tud/oas/responses/ErrorResponse.java
package org.tud.oas.responses;

import io.swagger.v3.oas.annotations.media.Schema;

@Schema(name = "ErrorResponse", description = """
        Error response containing the request endpoint and a error message.
        """)
public class ErrorResponse {
    @Schema(name = "request", description = """
            The request endpoint.
            """, example = "api/endpoint/action")
    public String request;

    @Schema(name = "error", description = """
            The error message.
            """, example = "invalid request")
    public String error;

    public ErrorResponse(String request, String message) {
        this.request = request;
        this.error = message;
    }
}
